### PR TITLE
Added tags and random test execution parameters

### LINKED ISSUE #50: Add support for tags (3.1.7 feature)
I would like to be able to declare tags of tests I [don't] want to run.

Semantically the same ability as:
select * from table(ut.run(a_tags => 'api,fast,-complex'));

Probably like:
`<configuration><tests><test><tags>api,fast,-complex</tags></test></tests></configuration>`

### PR DESCRIPTION
**Pull request by viniciusam**
Fixes #50 #42

### CODE REVIEW

**Comment by jgebal** [comment]
@pesse  - any chance for you to review this PR?

File: .travis.yml [hunk 1]
```diff
@@ -34,20 +34,19 @@ cache:
...
+  - bash .travis/install_utplsql.sh
   - bash .travis/install_demo_project.sh
-  - bash .travis/install_demo_owner_project.sh
 
 script:
   - echo "Run Unit Tests"
-  - mvn org.jacoco:jacoco-maven-plugin:prepare-agent test
+  - mvn test
   - echo "Run Integration Tests"
-  - mvn org.jacoco:jacoco-maven-plugin:prepare-agent verify -Pintegration -DdbUser="${DB_UT3_USER}" -DdbPass="${DB_UT3_PASS}" -DdbUrl="jdbc:oracle:thin:@${DB_URL}"
```
Comment (jgebal) [inline .travis.yml:45]: I think this is why we have no coverage on project anymore.
Any reason why `jacoco` was removed?

**Comment by jgebal** [comment]
@viniciusam  - something odd with sonar reporting - no coverage was reported on this PR.
If you look at develop - there was coverage on UtPLSQLMojo.java
https://sonarcloud.io/component_measures?id=org.utplsql%3Autplsql-maven-plugin&metric=coverage&selected=org.utplsql%3Autplsql-maven-plugin%3Asrc%2Fmain%2Fjava%2Forg%2Futplsql%2Fmaven%2Fplugin%2FUtPLSQLMojo.java&view=list

File: .travis.yml (hunk 1 above)
Comment (viniciusam) [reply .travis.yml:45]: It wasn't removed, it was added in the pom as a build step:
https://github.com/utPLSQL/utPLSQL-maven-plugin/blob/release/3.1.3/pom.xml#L252
https://github.com/utPLSQL/utPLSQL-maven-plugin/blob/release/3.1.3/pom.xml#L400

I remember testing this, I will need to take another look.

**Comment by jgebal** [comment]
It seems that coverage is not getting gathered on pull requests or something...

**Comment by jgebal** [comment]
I ran the MVN in debug mode.
Jacoco sonar import cannot find any Jacoco reports for some reason.
```
[INFO] 00:17:16.784 Sensor HTML [web] (done) | time=6ms
[INFO] 00:17:16.784 Sensor JaCoCo XML Report Importer [jacoco]
[DEBUG] 00:17:16.788 No reports found
[INFO] 00:17:16.788 Sensor JaCoCo XML Report Importer [jacoco] (done) | time=4ms
[INFO] 00:17:16.788 Sensor ThymeLeaf template sensor [securityjavafrontend]
[INFO] 00:17:16.789 Sensor ThymeLeaf template sensor [securityjavafrontend] (don
```
Full raw build log here:
https://api.travis-ci.org/v3/job/672308796/log.txt
or here:
https://travis-ci.org/github/utPLSQL/utPLSQL-maven-plugin/builds/672308795

**Comment by jgebal** [comment]
There is a lot of valuable code here.
I don't want lack of coverage report to be a blocker for merging this.
@pesse - We can merge this if you're OK.
Jacoco reporting on the project can be solved post-release - what do you think?

**Comment by pesse** [comment]
Yes, sure. I will try to get it running in a test setup, but current times are not that side-hobby friendly 😉

**Comment by viniciusam** [comment]
Thanks, I'm gonna merge it then. I have a branch with some code refactoring, so I might get the coverage up on that.

**Pull request merged**